File: fishy-coords-ui/src/Vessel.js
import React from 'react';
import { connect } from 'react-redux';
import { chooseVessel } from './actions';

import {
    ListGroup,
} from 'react-bootstrap';

class Vessel extends React.Component
{
    render()
    {
        let countryCode = "_unknown";
        let countryName = this.props.vessel.flag;

        let splitFlag =
           this.props.vessel.flag.split(' ');

        if (splitFlag.length === 2)
        {
            countryCode = splitFlag[0];
            countryName = splitFlag[1];
        }

        return <ListGroup.Item onClick={()=>
            this.props.chooseVessel(this.props.vessel)
        }>
            <div className="d-flex flex-column justify-content-between">
                <div className="d-flex flex-row justify-content-between">
                    <span style={{
                        width: "10em",
                    }}>
                        <span className="mr-1">
                            { countryCode }
                        </span>
                        <span>{ countryName }</span>
                    </span>
                    <span style={{flexGrow:1}}>
                        { this.props.vessel.name }
                    </span>
                </div>
                <div className="d-flex flex-row justify-content-between">
                    <span style={{
                        width: "10em",
                        fontWeight: 600
                    }}>
                        Registration Number
                    </span>
                    <span style={{flexGrow:1}}>
                        { this.props.vessel.registrationNumber }
                    </span>
                </div>
                <div className="d-flex flex-row justify-content-between">
                    <span style={{
                        width: "10em",
                        fontWeight: 600
                    }}>
                        IMO Number
                    </span>
                    <span style={{flexGrow:1}}>
                        { this.props.vessel.imoNumber }
                    </span>
                </div>
            </div>
        </ListGroup.Item>;
    }
}

export default connect(
    null,
    { chooseVessel }
)(Vessel);

File: fishyCoordsAPI/fishyCoordsAPI/Models/Vessel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace fishyCoordsAPI.Models
{
    public class Vessel
    {
        public string Id { get; set; }
        public string Name { get; set; }
        public string Flag { get; set; }
        public string RegistrationNumber { get; set; }
        public string IMONumber { get; set; }

        public Vessel()
        {
        }

        public Vessel(string id, string name, string flag, string regNumber, string imo)
        {
            Id = id;
            Name = name;
            Flag = flag;
            RegistrationNumber = regNumber;
            IMONumber = imo;
        }
    }
}
